src/kili/graphql/__init__.py

```diff
@@ -125,25 +123,24 @@ def execute_query_from_paginated_call(
                 as a value of the 'where' key in the global payload
             options: The query options
         """
-        payload: Dict[str, Any] = {"where": where_payload}
         if options.as_generator and not options.disable_tqdm:
             options._replace(disable_tqdm=True)
-        total_rows_queried = self.get_number_of_elements_to_query(where_payload, options)
+        total_rows_queried = self.get_number_of_elements_to_query(where, options)
         batch_size = min(100, options.first or 100)
 
         if total_rows_queried == 0:
             yield from ()
         else:
             with tqdm(total=total_rows_queried, disable=options.disable_tqdm) as pbar:
                 count_rows_retrieved = 0
-                while True:
+                while count_rows_retrieved < total_rows_queried:
                     skip = count_rows_retrieved + options.skip
-                    payload.update({"skip": skip, "first": batch_size})
+                    payload = {"where": where.graphql_payload, "skip": skip, "first": batch_size}
                     rows = api_throttle(self.client.execute)(query, payload)
                     rows = format_result("data", rows)
 
-                    if rows is None or len(rows) == 0:
-                        break
+                    # if rows is None or len(rows) == 0:
+                    #     break
```

Comment:
remove comments?

Reply:
I will add it back. It is useless but could still act as a safe line to keep